### PR TITLE
added character index

### PR DESCRIPTION
**Pull request by DWright89**
Index page for viewing a list of characters from the database has been added.  Includes api, seeders, migrations, and frontend components.

### CODE REVIEW

File: client/public/stylesheets/index.css
Comment (dovakeen118) [inline client/public/stylesheets/index.css]: I'd recommend using `.scss` rather than `.css` files for the added "superpowers"!

rather than a wildcard here, we should apply styles directly of put "globally applied" styles in the `main.scss` for better organization. Then styles related to the index page can go here

File: client/src/components/App.js
Comment (dovakeen118) [inline client/src/components/App.js]: You can remove the extra spaces `exact path="/characters"`

File: client/src/components/content/CharacterList.js
Comment (dovakeen118) [inline client/src/components/content/CharacterList.js:3]: We can add a space between library imports and our file imports

File: client/src/components/content/CharacterList.js
Comment (dovakeen118) [inline client/src/components/content/CharacterList.js]: You'll want to sync with your group on tab size indentation - this file has 4 tab size, while other files have 2. You can change this setting in VSCode so there is consistency between all team members in your project!

File: client/src/components/content/CharacterList.js
Comment (dovakeen118) [inline client/src/components/content/CharacterList.js]: formatting is a bit off on this component
```js
<CharacterTile
  key={character.id} // <- indented
/> // <- closing on new line
```

File: client/src/components/content/CharacterList.js
Comment (dovakeen118) [inline client/src/components/content/CharacterList.js:29]: We can remove empty lines

File: client/src/components/content/CharacterList.js
Comment (dovakeen118) [inline client/src/components/content/CharacterList.js]: this JSX can be indented, appearing nested under the `ol`

File: client/src/components/content/CharacterTile.js
Comment (dovakeen118) [inline client/src/components/content/CharacterTile.js]: Let's remove comments / unused code / empty lines

File: client/src/components/content/CharacterTile.js
Comment (dovakeen118) [inline client/src/components/content/CharacterTile.js]: Is this being used?

File: client/src/components/content/CharacterTile.js
Comment (dovakeen118) [inline client/src/components/content/CharacterTile.js:6]: The formatting can be updated here
```js
return ( 
  <li>
    {props.character.name}
  </li>
)
```

Comment (dovakeen118) [reply client/src/components/content/CharacterTile.js:6]: or all in one line
```js
return <li>{props.character.name}</li>
```

File: server/src/boot/environments/development.js
Comment (dovakeen118) [inline server/src/boot/environments/development.js]: While making other updates, let's remove the space that was added here at the beginning of the line so this file remains untouched

File: server/src/db/migrations/20220131193845_addCharacters.cjs
Comment (dovakeen118) [inline server/src/db/migrations/20220131193845_addCharacters.cjs]: Please remove!

File: server/src/db/migrations/seeders/characterSeeder.js
Comment (dovakeen118) [inline server/src/db/migrations/seeders/characterSeeder.js]: key value pairs should be indented within an object
```js
{
  key: value
}
```

File: server/src/models/Character.js
Comment (dovakeen118) [inline server/src/models/Character.js]: We should also add our other properties to ensure they have the correct type! It looks like we have all of the required fields accounted for, but also want to make sure that other things like "series" is a string, etc

File: server/src/routes/api/v1/characterRouter.js
Comment (dovakeen118) [inline server/src/routes/api/v1/characterRouter.js]: Let's remove empty space

File: server/src/routes/api/v1/characterRouter.js
Comment (dovakeen118) [inline server/src/routes/api/v1/characterRouter.js]: There should be spaces around `Character` in the destructuring
```js
{ Character }
```
https://github.com/airbnb/javascript#whitespace--in-braces

File: server/src/routes/api/v1/characterRouter.js
Comment (dovakeen118) [inline server/src/routes/api/v1/characterRouter.js]: Same whitespace here
```js
({ characters: characters })
```

We could also refactor this:
```js
({ characters })
```

File: server/src/routes/api/v1/characterRouter.js
Comment (dovakeen118) [inline server/src/routes/api/v1/characterRouter.js]: Added space around the errors

File: server/src/routes/clientRouter.js
Comment (dovakeen118) [inline server/src/routes/clientRouter.js]: We should add a space between comma separated values

Although you might decide to start listing the routes vertically as your app grows

```js
const clientRoutes = [
  "/", 
  "/user-sessions/new", 
  "/users/new",
  "/characters"
];
```

File: server/src/routes/rootRouter.js
Comment (dovakeen118) [inline server/src/routes/rootRouter.js]: There should be a space after the comma here

**Review by dovakeen118 — CHANGES_REQUESTED** [review]
Amazing work here! Most (if not all) of my comments are related to general styling best practices. Your model was missing validations for the rest of your properties that you'll want to take a look at. Some files have code comments, extra empty spaces, etc. to be cleaned up before merged in. Also take a look at your CSS styling, you'll want to avoid using that wildcard and apply more direct styling instead

File: client/src/assets/scss/main.scss
Comment (ohad-porat) [inline client/src/assets/scss/main.scss]: We really shouldn't be using the asterisk here, but rather apply these styles to either the `body` of the app or a specific element.

File: server/src/db/migrations/20220131193845_addCharacters.cjs
Comment (ohad-porat) [inline server/src/db/migrations/20220131193845_addCharacters.cjs:13]: Do we want `gameSeries` as not nullable or `gameTitle`? 
Previously on this PR, `gameTitle` was `.notNullable()`, but it seems like that was changed now

Comment (ohad-porat) [reply server/src/db/migrations/20220131193845_addCharacters.cjs:13]: Also, we want this to be inline with what we have in the model

File: server/src/models/Character.js
Comment (ohad-porat) [inline server/src/models/Character.js]: `gameTitle` is a required property here but not in the migration

File: server/src/db/migrations/20220131193845_addCharacters.cjs
Comment (ohad-porat) [inline server/src/db/migrations/20220131193845_addCharacters.cjs:14]: If we are adding a property in our migration we should use it in some way, especially if we are seeding data as well.
You should either remove `picutreUrl` or seed data at least in some of the records

File: server/src/models/Character.js
Comment (ohad-porat) [inline server/src/models/Character.js]: We can remove empty lines

File: server/src/routes/api/v1/characterRouter.js
Comment (ohad-porat) [inline server/src/routes/api/v1/characterRouter.js]: We should add a space between our key and value after the colon
```js
return res.status(500).json({ errors: err })
```

**Review by ohad-porat — CHANGES_REQUESTED** [review]
You did a great job addressing the feedback! 
I left some more feedback, mainly to sort out the discrepancy between your migration and model, and make a decision about `pictureUrl`

File: server/src/db/migrations/20220131193845_addCharacters.cjs
Comment (DWright89) [reply server/src/db/migrations/20220131193845_addCharacters.cjs:14]: Images were part of our plan from the beginning; they are used in the show page which has been added locally and is waiting for this merge to be completed before itself being PR'd

File: server/src/db/migrations/20220131193845_addCharacters.cjs
Comment (DWright89) [reply server/src/db/migrations/20220131193845_addCharacters.cjs:13]: We have addressed the model.  This was a mix-up on our end.

File: server/src/db/migrations/20220131193845_addCharacters.cjs
Comment (ohad-porat) [reply server/src/db/migrations/20220131193845_addCharacters.cjs:14]: That is great! I would encourage you to add some `pictureUrl`s to the seeder data.

**Review by ohad-porat — APPROVED** [review]

**Pull request merged**
